scijava/scijava-javadoc-parser/src/main/java/org/scijava/javadoc/parser/JavadocAnnotationProcessor.java

```diff
@@ -0,0 +1,195 @@
+        processingEnv.getMessager().printMessage(Diagnostic.Kind.NOTE, "Processing!!!!!");
+        this.yamlJavadocBuilder = new YamlJavadocBuilder(processingEnv);
+
+        final Map<String, String> options = processingEnv.getOptions();
+        final String packagesOption = options.get(PACKAGES_OPTION);
+
+        // Retain Javadoc for classes that match this predicate
+        final PackageFilter packageFilter =
+                packagesOption == null ? new PackageFilter() : new PackageFilter(packagesOption);
+
+        // Make sure each element only gets processed once.
+        final Set<Element> alreadyProcessed = new HashSet<>();
+
+        // If retaining Javadoc for all packages, the @RetainJavadoc annotation is redundant.
+        // Otherwise, make sure annotated classes have their Javadoc retained regardless of package.
+        if (!packageFilter.allowAllPackages()) {
+            for (TypeElement annotation : annotations) {
+                if (isRetainJavadocAnnotation(annotation)) {
+                    for (Element e : roundEnvironment.getElementsAnnotatedWith(annotation)) {
+                        generateJavadoc(e, alreadyProcessed);
+                    }
+                }
+            }
+        }
+
+        for (Element e : roundEnvironment.getRootElements()) {
+            if (packageFilter.test(e)) {
+                generateJavadoc(e, alreadyProcessed);
+            }
+        }
+
+        if (!roundEnvironment.getRootElements().isEmpty()) {
+            Element placeHolder = roundEnvironment.getRootElements().iterator().next();
+            for(String key: yamlMap.keySet()) {
+                try {
+                    outputYamlDoc(placeHolder, yaml.dump(yamlMap.get(key)), key);
+                }
+                catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private static boolean isRetainJavadocAnnotation(TypeElement annotation) {
+        return true;
+    }
+
+    private static <E extends Enum<E>> Optional<E> findValueOf(Class<E> enumClass, String valueName) {
+        try {
+            return Optional.of(Enum.valueOf(enumClass, valueName));
+        } catch (IllegalArgumentException e) {
+            return Optional.empty();
+        }
+    }
+
+    // ElementKind.RECORD was added in Java 16. We want to process records, but also
+    // remain compatible with earlier Java versions.
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    static final Optional<ElementKind> RECORD = findValueOf(ElementKind.class, "RECORD");
+
+    private static final EnumSet<ElementKind> elementKindsToInspect = EnumSet.of(
+            ElementKind.CLASS,
+            ElementKind.INTERFACE,
+            ElementKind.ENUM
+            // and RECORD, but only if the Java version supports it
+    );
+
+    static {
+        RECORD.ifPresent(elementKindsToInspect::add);
+    }
```

Comment:
I actually just removed the ability to scan records - we can add it back later if needed, since we aren't trying to be general in discovery anymore - only focused on ops